graphql/core/execution/parallel_execution.py

```diff
@@ -0,0 +1,287 @@
+import collections
+import functools
+
+from graphql.core import ExecutionResult, Source, parse, validate
+from graphql.core.defer import Deferred, DeferredDict, DeferredList, defer, fail, succeed
+from graphql.core.error import GraphQLError, format_error
+from graphql.core.execution import ExecutionContext, ResolveInfo, Undefined, collect_fields, default_resolve_fn, \
+    get_argument_values, get_field_def, get_operation_root_type
+from graphql.core.language import ast
+from graphql.core.type import GraphQLEnumType, GraphQLInterfaceType, GraphQLList, GraphQLNonNull, GraphQLObjectType, \
+    GraphQLScalarType, GraphQLUnionType
+from graphql.core.utils import is_nullish
+
+
+class Executor(object):
+    def __init__(self, schema, execution_middlewares=None):
+        self.execution_middlewares = execution_middlewares or []
+        self.schema = schema
+
+    def execute(self, request='', root=None, args=None, operation_name=None, execute_serially=False, validate_ast=True):
+        if not isinstance(request, ast.Document):
+            if not isinstance(request, Source):
+                request = Source(request, 'GraphQL request')
+
+            request = parse(request)
+
+        if validate_ast:
+            validation_errors = validate(self.schema, request)
+            if validation_errors:
+                return ExecutionResult(
+                    data=None,
+                    errors=validation_errors,
+                )
+
+        curried_execution_function = functools.partial(
+            self._execute_graphql_query,
+            root or object(),
+            request,
+            operation_name,
+            args or {},
+            execute_serially)
+
+        for middleware in self.execution_middlewares:
+            if hasattr(middleware, 'execution_result'):
+                curried_execution_function = functools.partial(middleware.execution_result, curried_execution_function)
+
+        return curried_execution_function()
+
+    def _execute_graphql_query(self, root, ast, operation_name, args, execute_serially=False):
+        ctx = ExecutionContext(self.schema, root, ast, operation_name, args)
+
+        d = defer(self._execute_operation, ctx, root, ctx.operation, execute_serially)
+        d.add_errback(
+            lambda error: ctx.errors.append(error)
```

Comment:
Since list.append() returns None, the return value of the error handler will then be sent to the next callback in the chain (in this case, Line `57`, which in turn will call `ExecutionResult(data, list(map(format_error, ctx.errors)))`  (but `data` is None!)